autograder/api/config.py

```diff
@@ -195,6 +221,34 @@ def get_argument_parser(
 
     return parser
 
+def _get_config_path(current_directory, config_file = DEFAULT_CONFIG_FILENAME, local_scope = None):
+    """
+    Search through the parent directories (until root) for a configuration file.
+    Stops at the first occurrence of the specified config file
+    (default: autograder.json) along the path to root.
+    Returns the path if a configuration file is found.
+    Otherwise, returns None.
+    """
+    current_path = pathlib.Path(current_directory)
+    config_file_path = os.path.join(current_path, config_file)
+
+    if (os.path.isfile(config_file_path)):
+        return config_file_path
+
+    if local_scope is not None:
+        local_scope_path = pathlib.Path(local_scope)
+        if local_scope_path == current_path:
+            return None
+
+    if current_path.parent == current_path:
+        return None
+
+    return _get_config_path(
```

Comment:
I feel like this can be simply handled in a loop instead of recursion.
Recursion is fun, but cases of trivial recursion should generally be handled with a loop instead (assuming the language does not optimize for tail recursion).